cmd/agent-js-worker/main.go

```diff
@@ -531,6 +552,35 @@ func agentOpts(opts *agentStartOpts) ([]aries.Option, error) {
 	return options, nil
 }
 
+// TODO: Use a persistent key instead of creating a new one every time this starts.
+
+// TODO: Use the KMS from the Aries framework for the EDV provider. Right now this isn't possible since
+//  the KMS in the Aries framework uses the provider passed in via WithStorageProvider for KMS storage.
+//  An EDV server can't use itself for KMS storage as this causes a chicken and egg problem.
+//  aries-framework-go will need to be updated to allow a different provider for just the KMS to resolve this.
```

Comment:
@Baha-sk

Reply:
I guess we need to separate the concepts of `secrets storage` and `document storage` in the framework.

Reply:
yes, we need a new storage type concept in the framework, probably `secure` and `open` storage types.

I think `document` storage type would confuse the underlying EDV storage, we need types that distinguish between secure storage that's used by EncryptedFormatter and open (insecure) storage used by everything else (the kms storage is an example, which technically uses a secretLock mechanism to secure it, but it's a different kind of `secure` storage)

Reply:
FYI indy-sdk just has [non-secrets](https://github.com/hyperledger/indy-sdk/blob/8a1bb3665d9d4927d978ee0b188f90fcee4d1fb7/libindy/src/commands/non_secrets.rs#L12) and everything else

Reply:
Technically private and symmetric keys are secrets too, but their storage in the kms have its own locking mechanism (not using EDV) in the framework.

The EncryptedFormatter needs the kms to create a proper secret storage. So the agent-sdk has two layers of secret storage (one for EDV used by EncryptedFormatter and one for the kms).